google-account-plugin/src/com/google/cloud/tools/intellij/stats/UsageTracker.java

```diff
@@ -33,4 +34,26 @@ void trackEvent(@NotNull String eventCategory,
       @Nullable String eventLabel,
       @Nullable Integer eventValue);
 
+  PartialTrackingEventAction trackEvent(String category);
+
+  interface SendsEvent {
+
+    void send();
+  }
+
+
+  interface PartialTrackingEventAction extends SendsEvent {
+
+    PartialTrackingEventLabel withAction(String value);
+
+    interface PartialTrackingEventLabel extends SendsEvent {
+
+      PartialTrackingEventValue andLabel(String value);
+
+      interface PartialTrackingEventValue extends SendsEvent {
```

Comment:
This is a strange-looking pattern I'm not aware of. What is this called?

In any case, we should make eventAction mandatory. Also, what I think ideal is that eventLabel/eventValue is a metadata pair and a user can provided many:

.addMetadata(key1, value1)
.addMetadata(key2, value2)

Currently, I settled down adding only one pair with the current `UsageTracker` interface. In that regard, changing the name `eventLabel` to `metadataKey` may be good.